src/state/state-service.ts

```diff
@@ -273,30 +274,37 @@ export default class StateService {
             throw new Error(`Can't located artefact ${artefactId} in edition ${editionId}`);
         }
 
-        const imagedObject = this._state.imagedObjects.find(artefact.imagedObjectId);
-        if (!imagedObject) {
-            console.error(`Can't locate imaged object ${artefact.imagedObjectId} for artefact ${artefact.id}`);
-            throw new Error(`Can't locate imaged object ${artefact.imagedObjectId} for artefact ${artefact.id}`);
-        }
+        const promises: Array<Promise<any>> = [];
+        let imagedObject: ImagedObject | undefined;
+        if (!artefact.isVirtual) {
+            imagedObject = this._state.imagedObjects.find(artefact.imagedObjectId);
+            if (!imagedObject) {
+                console.error(`Can't locate imaged object ${artefact.imagedObjectId} for artefact ${artefact.id}`);
+                throw new Error(`Can't locate imaged object ${artefact.imagedObjectId} for artefact ${artefact.id}`);
```

Comment:
Probably no need to throw an error, just run a GET on `v1/imaged-objects/{imagedObjectId}`, I doubt you would get an error then.

Reply:
We definitely need to throw an error here - it means the store is in an inconsistent state. When this code runs, the imaged object should have been preloaded into the state already.

Reply:
Is there a way to recover from this, or do you need to ask the user to reload?